milli/src/search/new/mod.rs

```diff
@@ -0,0 +1,494 @@
+            }
+            AscDesc::Asc(Member::Geo(point)) => {
+                if *geo_sorted {
+                    continue;
+                }
+                let geo_faceted_docids = ctx.index.geo_faceted_documents_ids(ctx.txn)?;
+                ranking_rules.push(Box::new(GeoSort::new(
+                    geo_strategy,
+                    geo_faceted_docids,
+                    point,
+                    true,
+                )?));
+            }
+            AscDesc::Desc(Member::Geo(point)) => {
+                if *geo_sorted {
+                    continue;
+                }
+                let geo_faceted_docids = ctx.index.geo_faceted_documents_ids(ctx.txn)?;
+                ranking_rules.push(Box::new(GeoSort::new(
+                    geo_strategy,
+                    geo_faceted_docids,
+                    point,
+                    false,
+                )?));
+            }
+        };
+    }
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn execute_search(
+    ctx: &mut SearchContext,
+    query: &Option<String>,
+    terms_matching_strategy: TermsMatchingStrategy,
+    exhaustive_number_hits: bool,
+    filters: &Option<Filter>,
+    sort_criteria: &Option<Vec<AscDesc>>,
+    geo_strategy: geo_sort::Strategy,
+    from: usize,
+    length: usize,
+    words_limit: Option<usize>,
+    placeholder_search_logger: &mut dyn SearchLogger<PlaceholderQuery>,
+    query_graph_logger: &mut dyn SearchLogger<QueryGraph>,
+) -> Result<PartialSearchResult> {
+    let mut universe = if let Some(filters) = filters {
+        filters.evaluate(ctx.txn, ctx.index)?
+    } else {
+        ctx.index.documents_ids(ctx.txn)?
+    };
+
+    check_sort_criteria(ctx, sort_criteria.as_ref())?;
+
+    let mut located_query_terms = None;
+
+    let query_terms = if let Some(query) = query {
+        // We make sure that the analyzer is aware of the stop words
+        // this ensures that the query builder is able to properly remove them.
+        let mut tokbuilder = TokenizerBuilder::new();
+        let stop_words = ctx.index.stop_words(ctx.txn)?;
+        if let Some(ref stop_words) = stop_words {
+            tokbuilder.stop_words(stop_words);
+        }
+
+        let script_lang_map = ctx.index.script_language(ctx.txn)?;
+        if !script_lang_map.is_empty() {
+            tokbuilder.allow_list(&script_lang_map);
+        }
+
+        let tokenizer = tokbuilder.build();
+        let tokens = tokenizer.tokenize(query);
+
+        let query_terms = located_query_terms_from_string(ctx, tokens, words_limit)?;
+        if query_terms.is_empty() {
+            // Do a placeholder search instead
+            None
+        } else {
+            Some(query_terms)
+        }
+    } else {
+        None
+    };
+    let bucket_sort_output = if let Some(query_terms) = query_terms {
+        let graph = QueryGraph::from_query(ctx, &query_terms)?;
```

Comment:
important for highlighter: @loiclec return locatedqueryterms containing the ngrams